File: gdk/cocos-creator/assets/fearth-gdk/scripts/data/LoginResponse.ts
import { _decorator } from "cc";

const { ccclass } = _decorator;

@ccclass('LoginResponse')
export class LoginResponse {
    public success: boolean = true;
    public errorCode: number = 0;
    public userId: string = "";
    public token: string = "";

    public static error(errorCode: number): LoginResponse {
        var response = new LoginResponse();
        response.errorCode = errorCode;
        return response;
    }
}

File: gdk/android/app/src/main/java/com/fearth/gdk/data/LoginResponse.java
package com.fearth.gdk.data;

public class LoginResponse {
    public boolean success = false;
    public int errorCode = 0;
    public String userId = null;
    public String token = null;

    public static LoginResponse error(int errorCode) {
        LoginResponse response = new LoginResponse();
        response.errorCode = errorCode;
        return response;
    }
}
